src/Http/Http/src/Features/RequestCookiesFeature.cs

```diff
@@ -88,7 +84,7 @@ public IRequestCookieCollection Cookies
                 {
                     if (_parsedValues == null || _parsedValues.Count == 0)
                     {
-                        HttpRequestFeature.Headers.Remove(HeaderNames.Cookie);
+                        HttpRequestFeature.Headers.Cookie = default;
```

Comment:
This is interesting. This is the behavior we have today? Makes you wonder if these should be nullable.

Reply:
Yep; its setting it to a `StringValues.Length == 0` https://github.com/dotnet/aspnetcore/blob/0aae147c4b8b7ecf0431c076f77892588e7ea0ce/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.cs#L63-L70

Reply:
Guessing this is a kestrel only behavior and the more natural thing to do is to call Remove(..)

Reply:
Kestrel has to check due to the internal implementation where it maintains flags saying if the header is set or not; so it needs to intercept and change the flags if you blank it.

However what's the observable difference between a collection that returns `StringValues.Empty`/`default` for every unset header and setting a header to `StringValues.Empty`/`default`? They are in up in the same state?

Can still use `Headers.Remove(HeaderNames.Cookie)` if desired

Alas can't add an extension to a property and don't think this would work
```csharp
public static class HeaderExtensions
{
    public static void Remove(ref StringValues value)
    {
        value = default;
    }
}
```
To give `Headers.Cookie.Remove()`

Reply:
IIS/HttpSys has some inconsistencies here around its internal flag setting (doesn't clear like Kestrel); however it then uses `.Count` to say whether the header is there; so may amount to the same
https://github.com/dotnet/aspnetcore/blob/0aae147c4b8b7ecf0431c076f77892588e7ea0ce/src/Shared/HttpSys/RequestProcessing/RequestHeaders.Generated.cs#L1369-L1372

Reply:
And the reference implementation `HeaderDictionary` removes if set to `default`
https://github.com/dotnet/aspnetcore/blob/0aae147c4b8b7ecf0431c076f77892588e7ea0ce/src/Http/Http/src/HeaderDictionary.cs#L89-L97

Reply:
I totally agree as `.Remove()` being the more natural API, this one struck me as weird almost enough to comment earlier.

Reply:
Could have a `Remove` struct with a method for each header? 😅
```csharp
Headers.Remove.Cookie()
```

Reply:
hmmm I just checked https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md and there doesn't seem to be an ethics clause added yet, so I guess that API is okay.